### PR TITLE
SERF

### PR DESCRIPTION
**Pull request by cryptoeejit**
The amazing SERF community coin.

### CODE REVIEW

File: src/Networks/Blockcore.Networks.SERF/Networks/Deployments/SERFBIP9Deployments.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      /// <param name="deployment">The deployment number.</param>
+      /// <returns>The deployment flags.</returns>
+      public override BIP9DeploymentFlags GetFlags(int deployment)
+      {
+         var flags = new BIP9DeploymentFlags();
+
+         switch (deployment)
+         {
+            case ColdStaking:
+               flags.ScriptFlags |= ScriptVerify.CheckColdStakeVerify;
```
Comment (sondreb) [inline src/Networks/Blockcore.Networks.SERF/Networks/Deployments/SERFBIP9Deployments.cs:40]: Don't need the append flag here, can simply be =

File: src/Networks/Blockcore.Networks.SERF/Networks/Deployments/SERFBIP9Deployments.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace Blockcore.Networks.SERF.Deployments
+{
+   /// <summary>
+   /// BIP9 deployments for the Blockcore sample coin network.
+   /// </summary>
+   public class SERFBIP9Deployments : BIP9DeploymentsArray
+   {
+      // The position of each deployment in the deployments array. Note that this is decoupled from the actual position of the flag bit for the deployment in the block version.
+      public const int TestDummy = 0;
```
Comment (sondreb) [inline src/Networks/Blockcore.Networks.SERF/Networks/Deployments/SERFBIP9Deployments.cs:13]: Remove the dummy and reorganize the indexes of deployments.

File: src/Networks/Blockcore.Networks.SERF/Networks/Deployments/SERFBIP9Deployments.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      public const int CSV = 1;
+      public const int Segwit = 2;
+      public const int ColdStaking = 3;
+
+      // The number of deployments.
+      public const int NumberOfDeployments = 4;
```
Comment (sondreb) [inline src/Networks/Blockcore.Networks.SERF/Networks/Deployments/SERFBIP9Deployments.cs:19]: Make this 3 when you remove the dummy

File: src/Networks/Blockcore.Networks.SERF/Networks/SERFRegTest.cs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+             consensusFactory: consensusFactory,
+             consensusOptions: consensusOptions,
+             coinType: setup.CoinType,
+             hashGenesisBlock: genesisBlock.GetHash(),
+             subsidyHalvingInterval: 210000,
+             majorityEnforceBlockUpgrade: 750,
+             majorityRejectBlockOutdated: 950,
+             majorityWindow: 1000,
+             buriedDeployments: buriedDeployments,
+             bip9Deployments: new NoBIP9Deployments(),
```
Comment (sondreb) [inline src/Networks/Blockcore.Networks.SERF/Networks/SERFRegTest.cs:80]: You should consider copying the same configuration from main.

File: src/Networks/Blockcore.Networks.SERF/SERFSetup.cs [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+         Magic = "01-4F-4E-4D",
+         CoinType = 712, // SLIP-0044: https://github.com/satoshilabs/slips/blob/master/slip-0044.md,
+         PremineReward = 1000000,
+         PoWBlockReward = 8,
+         PoSBlockReward = 2,
+         LastPowBlock = 10000000,
+         GenesisText = "15-06-1215 - JOHN, by the grace of God King of England", // The New York Times, 2020-04-16
+         TargetSpacing = TimeSpan.FromSeconds(120),
+         ProofOfStakeTimestampMask = 0x0000000F, // 0x0000003F // 64 sec
+         PoSVersion = 3
```
Comment (sondreb) [inline src/Networks/Blockcore.Networks.SERF/SERFSetup.cs:27]: You should consider going with POSv4 right away

**Review by sondreb — CHANGES_REQUESTED** [review]

**Comment by cryptoeejit** [comment]
Thanks.  Will get to it.

**Comment by cryptoeejit** [comment]
Think I need to review the setup more,  Hopefully resubmit when its better,

**Pull request closed without merging**